components/external-services-mock/cmd/main.go

```diff
@@ -56,9 +71,40 @@ func main() {
 		log.Fatal(err)
 	}
 
-	log.Printf("External Services Mock up and running on address: %s", cfg.Address)
-	err = http.ListenAndServe(cfg.Address, handler)
-	exitOnError(err, "while running up http server")
+	certSecuredHandler, err := initCertSecuredAPIs(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	server := &http.Server{
+		Addr:    cfg.Address,
+		Handler: handler,
+	}
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM([]byte(cfg.CACert))
+	if len(cfg.CertSvcRootCA) > 0 {
+		rootCA, err := base64.StdEncoding.DecodeString(cfg.CertSvcRootCA)
+		exitOnError(err, "while base64 decoding Cert Svc ROOT CA")
+		caCertPool.AppendCertsFromPEM(rootCA)
+	}
+
+	certSecuredServer := &http.Server{
+		Addr:    cfg.CertSecuredServerAddress,
+		Handler: certSecuredHandler,
+		TLSConfig: &tls.Config{
+			ClientCAs:  caCertPool,
+			ClientAuth: tls.RequireAndVerifyClientCert,
+		},
+	}
+
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
+
+	go startServer(ctx, server, wg)
+	go startServer(ctx, certSecuredServer, wg)
```

Comment:
The secured server should be started with `ListenAndServerTLS` in order for the client cert authentication to work